grpc/interceptors_test.go

```diff
@@ -136,7 +136,7 @@ func (s *testServer) Chill(ctx context.Context, in *test_proto.Time) (*test_prot
 		spent := int64(time.Since(start) / time.Nanosecond)
 		return &test_proto.Time{Time: spent}, nil
 	case <-ctx.Done():
-		return nil, status.Errorf(codes.DeadlineExceeded, "the chiller overslept")
+		return nil, status.Errorf(codes.Canceled, "the chiller overslept")
```

Comment:
Neither the old nor the new code was totally correct. We can get a read off of ctx.Done() for _either_ canceled _or_ deadline exceeded reasons. In this test we happen to know it's for deadline exceeded reasons because we set a timeout in the test runner, and never run `cancel()` until _after_ `Chill()` returns.

However, if we want to be more correct this should probably be:

```suggestion
		return nil, ctx.Err()
```

Reply:
The issue isn't simply that the error returned here was "wrong" -- if anything, it was closer to right, because `ctx.Err()` is also DeadlineExceeded.

The issue is that returning DeadlineExceeded from within the server is (to the client) indistinguishable from the client returning DeadlineExceeded itself because the server took too long. Then that DeadlineExceeded gets converted by the client interceptor into a `deadlineDetails`, which produces the error message "Chiller.Chill timed out after 0ms". (The 0ms is because time during unittests is fake.) This error message is identical to what the other, shorter, test cases are looking for. Therefore I changed the error returned by Chiller.Chill to something else (Canceled), so that it could still be distinguished by the test.

So unfortunately switching this to ctx.Err() makes the test cases indistinguishable again.

Reply:
I've changed this to a errors.New() with a self-documenting error message to help make this test easier to understand.